### PR TITLE
feat: `dx.indicator` spec

### PR DESCRIPTION
**Pull request by jnumainville**
spec for #1019 

I haven't really written a spec for `dx` before but it made sense to me to just put it in a new function with `NotImplementedError`.

Instead of just enabling one indicator, you can create multiple at one time with a plot by that makes subplots of indicators. This enables, for example, the creation of a new indicator automatically as a new symbol ticks in.  
The default behavior is only a `number` if you only provide `value` and `number+delta` if you provide `reference` as well.
https://plotly.com/python/indicator/

Syntax is otherwise generally in line with other `dx` functions and most other behavior is fairly straightforward.

### CODE REVIEW

**Review by mofojed — COMMENTED** [review]
Makes sense to me... only thing I'd like is to be able to pass in a value that I have outside of a table as well. Calculated from a user input or something.
It could be that we pump it through a table anyway, but I'd like to see an example of how to hook that up.

**Comment by jnumainville** [comment]
> Makes sense to me... only thing I'd like is to be able to pass in a value that I have outside of a table as well. Calculated from a user input or something. It could be that we pump it through a table anyway, but I'd like to see an example of how to hook that up.

Added an example of putting some values in a table.
I think if we want something more comprehensive, I'd argue it's beyond the scope of this initial indicator implementation because what we would really want to do is rework what we allow passed into `table` for all plots. Things like `dicts`, `lists`, etc. I don't really like the idea of breaking the abstraction by passing in values directly to parameters that take columns as it adds complexity and inconsistency, but I'd be open to expanding what we allow passed into `table`.

**Review by mofojed — DISMISSED** [review]

**Review by margaretkennedy — APPROVED** [review]

File: plugins/plotly-express/docs/indicator.md [hunk 1]
```diff
@@ -0,0 +1,273 @@
+# Indicator Plot
+
+An indicator plot is a type of plot that highlights a collection of numeric values.
+
+### What are indicator plots useful for?
+
+- **Highlight specific metrics**: Indicator plots are useful when you want to highlight specific numeric metrics in a visually appealing way.
+- **Compare metrics to a reference value**: Indicator plots are useful to compare metrics to a reference value, such as a starting value or a target value.
+- **Compare metrics to each other**: Indicator plots are useful to compare multiple metrics to each other by highlighting where they fall relative to each other.
+-
```
Comment (dsmmcken) [inline plugins/plotly-express/docs/indicator.md:10]: stray dash

Comment (jnumainville) [reply plugins/plotly-express/docs/indicator.md]: removed

**Review by margaretkennedy — APPROVED** [review]

**Review by dsmmcken — APPROVED** [review]

**Review by mofojed — APPROVED** [review]
Looks good to me.

This has inspired me to create a utility function called `dict_table` for where you just want a table from a dictionary, e.g.
```python
from deephaven import dict_table

my_table = dict_table({
  "MyValue": [my_value],
  "MyReference": [my_reference],
})
```

Or even just assume if it's not a list passed in, that it is just a one row table and you don't need to pass in an array explicitly, e.g.:
```python

from deephaven import dict_table

my_table = dict_table({
  "MyValue": my_value,
  "MyReference": my_reference,
})
```

Don't have as control over types ... but it would be nice for creating quick tables like in this case.

**Pull request merged**